spock-core/src/main/java/org/spockframework/runtime/GroovyRuntimeUtil.java

```diff
@@ -301,4 +302,9 @@ public static Method toMethod(@Nullable MetaMethod metaMethod) {
     CachedMethod cachedMethod = ObjectUtil.asInstance(metaMethod, CachedMethod.class);
     return cachedMethod == null ? null : cachedMethod.getCachedMethod();
   }
+
+  @Internal
+  public static boolean isGroovy2() {
```

Comment:
Add another method `isGroovy3()` and maybe initialize those values in static final constants

Reply:
I was already thinking about Groovy 4, but it can take some time to get it ;-).
A major drawback of `isGroovy3()` is that we might with no test executed. Maybe `isNotGroovy2()`?

Reply:
I chose the last proposal and added `isGroovy3()`.